ddtrace/tracer/metrics.go

```diff
@@ -8,38 +8,54 @@ package tracer
 import (
 	"runtime"
 	"runtime/debug"
+	"sync/atomic"
 	"time"
 
-	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
 )
 
 // defaultMetricsReportInterval specifies the interval at which runtime metrics will
 // be reported.
 const defaultMetricsReportInterval = 10 * time.Second
 
-type gauger interface {
+type statsdClient interface {
+	Incr(name string, tags []string, rate float64) error
+	Count(name string, value int64, tags []string, rate float64) error
 	Gauge(name string, value float64, tags []string, rate float64) error
+	Timing(name string, value time.Duration, tags []string, rate float64) error
+	Close() error
 }
 
-// reportMetrics periodically reports go runtime metrics to the specified gauger at
+type noopStatsdClient struct{}
```

Comment:
Can we replace this with https://godoc.org/github.com/DataDog/datadog-go/statsd#NoOpClient ?